src/sentry/ratelimits/redis.py

```diff
@@ -23,24 +47,39 @@ def validate(self):
             raise InvalidConfiguration(str(e))
 
     def is_limited(self, key, limit, project=None, window=None):
-        if window is None:
-            window = self.window
+        is_limited, _ = self.is_limited_with_value(key, limit, project=project, window=window)
+        return is_limited
 
-        key_hex = md5_text(key).hexdigest()
-        bucket = int(time() / window)
+    def current_value(self, key: int, project: Project = None, window: int = None) -> int:
+        redis_key = self._construct_redis_key(key, project=project, window=window)
 
-        if project:
-            key = f"rl:{key_hex}:{project.id}:{bucket}"
-        else:
-            key = f"rl:{key_hex}:{bucket}"
+        try:
+            with self.cluster.map() as client:
+                result = client.get(redis_key)
+            current_count = result.value
+            if current_count is None:
+                # Key hasn't been created yet, therefore no hits done so far
+                return 0
+            return int(current_count)
+        except RedisError as e:
+            # Don't report any existing hits when there is a redis error.
+            # Log what happened and move on
+            capture_exception(e)
```

Comment:
Nit: I'd typically use `logger.exception('Failed to retrieve current value from redis')` here. That way you have a descriptive message and the exception is captured automatically.